File: basemodule/src/main/java/com/android/base/mmodel/RegisterLoginData.java
package com.android.base.mmodel;

import java.util.List;


public class RegisterLoginData extends BaseData {

   public Data data;

    public static class Data {
        public String username;
        public String password;
        public String email, icon, token;
        public int id, type;
        //    private List<Integer> chapterTops;
        public List<Integer> collectIds;

        @Override
        public String toString() {
            return "RegisterLoginData{" +
                    "username='" + username + '\'' +
                    ", password='" + password + '\'' +
                    ", email='" + email + '\'' +
                    ", icon='" + icon + '\'' +
                    ", token='" + token + '\'' +
                    ", id=" + id +
                    ", type=" + type +
                    ", collectIds=" + collectIds +
                    '}';
        }
    }
}

File: basemodule/src/main/java/com/android/learn/base/mmodel/RegisterLoginData.kt
package com.android.learn.base.mmodel

/**
 * @author quchao
 * @date 2018/2/26
 */

class RegisterLoginData : BaseData() {

    var data: Data? = null

    inner class Data {
        lateinit var username: String
        lateinit var password: String
        lateinit var email: String
        lateinit var icon: String
        lateinit var token: String
        var id: Int = 0
        var type: Int = 0
        //    private List<Integer> chapterTops;
        lateinit var collectIds: List<Int>

        override fun toString(): String {
            return "RegisterLoginData{" +
                    "username='" + username + '\''.toString() +
                    ", password='" + password + '\''.toString() +
                    ", email='" + email + '\''.toString() +
                    ", icon='" + icon + '\''.toString() +
                    ", token='" + token + '\''.toString() +
                    ", id=" + id +
                    ", type=" + type +
                    ", collectIds=" + collectIds +
                    '}'.toString()
        }
    }
}
